File: server/config/gcp_config.py
import os
from google.cloud import secretmanager

# GCP Project configuration
GCP_PROJECT_ID = os.environ.get('GCP_PROJECT_ID')
GCP_REGION = os.environ.get('GCP_REGION', 'us-central1')

# Cloud SQL (PostgreSQL) configuration
CLOUD_SQL_INSTANCE_NAME = os.environ.get('CLOUD_SQL_INSTANCE_NAME')
CLOUD_SQL_DATABASE_NAME = os.environ.get('CLOUD_SQL_DATABASE_NAME')
CLOUD_SQL_USER = os.environ.get('CLOUD_SQL_USER')
CLOUD_SQL_PASSWORD = os.environ.get('CLOUD_SQL_PASSWORD')

# Cloud Memorystore (Redis) configuration
REDIS_HOST = os.environ.get('REDIS_HOST')
REDIS_PORT = os.environ.get('REDIS_PORT', 6379)

# Cloud Firestore configuration
FIRESTORE_COLLECTION = os.environ.get('FIRESTORE_COLLECTION', 'adapt_agent_data')

# Vertex AI configuration
VERTEX_AI_ENDPOINT = os.environ.get('VERTEX_AI_ENDPOINT')

def get_secret(secret_id):
    client = secretmanager.SecretManagerServiceClient()
    name = f"projects/{GCP_PROJECT_ID}/secrets/{secret_id}/versions/latest"
    response = client.access_secret_version(request={"name": name})
    return response.payload.data.decode("UTF-8")

# Use Secret Manager for sensitive information
if not CLOUD_SQL_PASSWORD:
    CLOUD_SQL_PASSWORD = get_secret('cloud-sql-password')

# Database URL for SQLAlchemy
DATABASE_URL = f"postgresql://{CLOUD_SQL_USER}:{CLOUD_SQL_PASSWORD}@/{CLOUD_SQL_DATABASE_NAME}?host=/cloudsql/{GCP_PROJECT_ID}:{GCP_REGION}:{CLOUD_SQL_INSTANCE_NAME}"

# Redis URL
REDIS_URL = f"redis://{REDIS_HOST}:{REDIS_PORT}"

# Firestore client initialization
from google.cloud import firestore
db = firestore.Client(project=GCP_PROJECT_ID)

# Vertex AI client initialization
from google.cloud import aiplatform
aiplatform.init(project=GCP_PROJECT_ID, location=GCP_REGION)

File: server/config/gcp_config.js
require('dotenv').config();

async function getGcpConfig() {
  if (process.env.NODE_ENV === 'development') {
    return {
      DATABASE_URL: `postgresql://${process.env.DB_USER}:${process.env.DB_PASSWORD}@${process.env.DB_HOST}:${process.env.DB_PORT}/${process.env.DB_NAME}`,
      REDIS_URL: 'redis://localhost:6379',
      GCP_PROJECT_ID: 'development',
      GCP_REGION: 'development',
      CLOUD_SQL_INSTANCE_NAME: 'development',
      CLOUD_SQL_DATABASE_NAME: process.env.DB_NAME,
      CLOUD_SQL_USER: process.env.DB_USER,
      CLOUD_SQL_PASSWORD: process.env.DB_PASSWORD,
      REDIS_HOST: 'localhost',
      REDIS_PORT: 6379,
      FIRESTORE_COLLECTION: 'development',
      VERTEX_AI_ENDPOINT: 'development'
    };
  }

  // Production GCP configuration (unchanged)
  const { SecretManagerServiceClient } = require('@google-cloud/secret-manager');
  const secretClient = new SecretManagerServiceClient();

  async function accessSecret(secretName) {
    try {
      const [version] = await secretClient.accessSecretVersion({
        name: `projects/${process.env.GCP_PROJECT_ID}/secrets/${secretName}/versions/latest`,
      });
      return version.payload.data.toString();
    } catch (error) {
      console.error(`Error accessing secret ${secretName}:`, error);
      return null;
    }
  }

  const [
    GCP_PROJECT_ID,
    GCP_REGION,
    CLOUD_SQL_INSTANCE_NAME,
    CLOUD_SQL_DATABASE_NAME,
    CLOUD_SQL_USER,
    CLOUD_SQL_PASSWORD,
    REDIS_HOST,
    REDIS_PORT,
    FIRESTORE_COLLECTION,
    VERTEX_AI_ENDPOINT
  ] = await Promise.all([
    accessSecret('GCP_PROJECT_ID'),
    accessSecret('GCP_REGION'),
    accessSecret('CLOUD_SQL_INSTANCE_NAME'),
    accessSecret('CLOUD_SQL_DATABASE_NAME'),
    accessSecret('CLOUD_SQL_USER'),
    accessSecret('CLOUD_SQL_PASSWORD'),
    accessSecret('REDIS_HOST'),
    accessSecret('REDIS_PORT'),
    accessSecret('FIRESTORE_COLLECTION'),
    accessSecret('VERTEX_AI_ENDPOINT')
  ]);

  const DATABASE_URL = `postgresql://${CLOUD_SQL_USER}:${CLOUD_SQL_PASSWORD}@/${CLOUD_SQL_DATABASE_NAME}?host=/cloudsql/${GCP_PROJECT_ID}:${GCP_REGION}:${CLOUD_SQL_INSTANCE_NAME}`;
  const REDIS_URL = `redis://${REDIS_HOST}:${REDIS_PORT}`;

  return {
    GCP_PROJECT_ID,
    GCP_REGION,
    CLOUD_SQL_INSTANCE_NAME,
    CLOUD_SQL_DATABASE_NAME,
    CLOUD_SQL_USER,
    CLOUD_SQL_PASSWORD,
    REDIS_HOST,
    REDIS_PORT,
    FIRESTORE_COLLECTION,
    VERTEX_AI_ENDPOINT,
    DATABASE_URL,
    REDIS_URL
  };
}

module.exports = getGcpConfig;